services/web/server/src/simcore_service_webserver/exporter/formatters/sds/xlsx/templates/utils.py

```diff
@@ -0,0 +1,38 @@
+from typing import List, Dict, Any, Type
+from pydantic import BaseModel
+from openpyxl.utils import get_column_letter
+
+
+def ensure_same_field_length(
+    fields_to_check: List[str], values: Dict[str, Any]
+) -> None:
+    field_lengths = {field: values.get(field, -1) for field in fields_to_check}
+    # expecting one single entry in a set if all lengths are the same
+    if len(set(field_lengths.values())) != 1:
+        raise ValueError(
+            f"Not all fields had the same length, please check {field_lengths}"
+        )
+
+
+def ensure_correct_instance(
```

Comment:
you could have implemented this as a decorator that checks specific params? :-)

Reply:
Yes, but I wanted to skip some pain 😭